Arduino Uno R3/libraries/ADE9153A/ADE9153AAPI.h

```diff
@@ -0,0 +1,184 @@
+  BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT 
+  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED 
+  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR 
+  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF 
+  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **************************************************************************/
+#ifndef ADE9153AAPI_h
+#define ADE9153AAPI_h
+
+#include "Arduino.h"
+#include "ADE9153A.h"
+
+/**************************************************************************
+  Definitions
+ **************************************************************************/
+ /*Configuration Registers*/
+#define ADE9153A_AI_PGAGAIN 0x000A		/*Signal on IAN, current channel gain=16x*/
+#define ADE9153A_CONFIG0 0x00000000		/*Datapath settings at default*/
+#define ADE9153A_CONFIG1 0x0300			/*Chip settings at default*/
+#define ADE9153A_CONFIG2 0x0C00			/*High-pass filter corner, fc=0.625Hz*/
+#define ADE9153A_CONFIG3 0x0000			/*Peak and overcurrent settings*/
+#define ADE9153A_ACCMODE 0x0010			/*Energy accumulation modes*/
+										/*Bit 4, 0 for 50Hz, 1 for 60Hz*/
+
+#define ADE9153A_VLEVEL 0x002C11E8		/*Assuming Vnom=1/2 of fullscale*/
+#define ADE9153A_ZX_CFG 0x0000			/*ZX low-pass filter select*/
+#define ADE9153A_MASK 0x00000100		/*Enable EGYRDY interrupt*/
+
+#define ADE9153A_ACT_NL_LVL 0x000033C8
+#define ADE9153A_REACT_NL_LVL 0x000033C8
+#define ADE9153A_APP_NL_LVL 0x000033C8
+
+ /*Constant Definitions*/
+#define ADE9153A_RUN_ON 0x0001			/*DSP On*/
+#define ADE9153A_COMPMODE 0x0005		/*Initialize for proper operation*/
+
+#define ADE9153A_VDIV_RSMALL 0x03E8		/*Small resistor on board is 1kOhm=0x3E8*/
+ 
+ /*Energy Accumulation Settings*/
+#define ADE9153A_EP_CFG 0x0009			/*Energy accumulation configuration*/
+#define ADE9153A_EGY_TIME 0x0F9F		/*Accumulate energy for 4000 samples*/
+
+ /*Temperature Sensor Settings*/
+#define ADE9153A_TEMP_CFG 0x000C		/*Temperature sensor configuration*/
+
+ /*Ideal Calibration Values for ADE9153A Shield Based on Sensor Values*/
+#define CAL_IRMS_CC		0.838190	// (uA/code)
+#define CAL_VRMS_CC		13.41105	// (uV/code)	
+#define CAL_POWER_CC 	1508.743	// (uW/code) Applicable for Active, reactive and apparent power
+#define CAL_ENERGY_CC	0.858307	// (uWhr/xTHR_HI code)Applicable for Active, reactive and apparent energy
+
+/**************************************************************************
+  Structures and Global Variables
+ **************************************************************************/
+ 
+typedef struct EnergyRegs

```

Comment:
not using typedef makes it more descriptive when declaring variables. It is more verbose, but you understand what that variable is;

EnergyRegs a;

struct EnergyRegs a; - from this I understand that this is a structure